design/src/rfcs/rfc0023_serialization_and_deserialization.md

```diff
@@ -0,0 +1,295 @@
+| cargo build --release --features unstable-serde-deserialize | 1m0.107s  | 5m10.231s | 0m11.699s |
+| cargo build --all-features                                  | 0m48.959s | 2m45.688s | 0m13.359s |
+| cargo build --release --all-features                        | 1m3.198s  | 5m26.076s | 0m12.311s |
+
+- `aws-sdk-ec2`
+ 
+| command                                                     | real time | user time  | sys time  |
+| ----------------------------------------------------------- | --------- | ---------- | --------- |
+| cargo build                                                 | 1m20.041s | 2m14.592s  | 0m6.611s  |
+| cargo build --release                                       | 2m29.480s | 9m19.530s  | 0m15.957s |
+| cargo build --features unstable-serde-serialize             | 2m0.555s  | 4m24.881s  | 0m16.131s |
+| cargo build --release --features unstable-serde-serialize   | 2m45.002s | 9m43.098s  | 0m16.886s |
+| cargo build --features unstable-serde-deserialize           | 3m10.857s | 5m34.246s  | 0m18.844s |
+| cargo build --release --features unstable-serde-deserialize | 3m47.531s | 10m52.017s | 0m18.404s |
+| cargo build --all-features                                  | 3m31.473s | 6m1.052s   | 0m19.681s |
+| cargo build --release --all-features                        | 3m45.208s | 8m46.168s  | 0m10.211s |
+
+
+## Misleading Results
+SDK team previously expressed concern that serialized data may be misleading.  
+We believe that features implemented as part of this RFC does not produce misleading result as we focus on builder types and it's corresponding data types which are mapped to serde's data type model with the derive macro.  
+
+# Feature Gate
+`Serde` traits are implemented behind feature gates. 
+`Serialize`  is implemented behind `serialize`., while `Deserialize` is implemented behind  `deserialize`.
+Users must enable `unstable` feature to expose those features.
+
+We considered giving each feature a dedicated feature gate such as `unstable-serde-serialize`. 
+In this case, we will need to change the name of feature gates entirely once it leaves the unstable status which will cause users to make changes to their code base.
+We conclude that this brings no benefit to the users.
+
+## Keeping both features behind the same feature gate
+We considered keeping both features behind the same feature gate.  
+There is no significant difference in the complexity of implementation.
+We do not see any benefit in keeping them behind a same feature-gate as this will only result in increase of compile time when users are not in need of one of the feature.
+
+## Different feature gates for different data types
+We considered implementing different feature gates for output, input and their corresponding data types.
+For example, output and input types can have `output-serde-*` and `input-serde-*`.
+We are unable to do this as relevant meta data is not available during the code-gen.
```

Comment:
We may be able to do this in a later feature anyways, so I think it's totally fine to avoid defining this.